crud-app/app/Http/Controllers/ProductController.php

```diff
@@ -15,10 +15,18 @@ class ProductController extends Controller
      * 
      * @return \Illuminate\Http\Response
      */
-    public function index(): Response
+    public function index(Request $request): Response
     {
-        $products = Product::paginate(10);
-        return response(view('products.index', ['products' => $products]));
+        // Assigning variables to sort by 'name' in ascending order in our sql call below
+        $sortBy = $request->query('sortBy') ?? 'name';
+        $direction = $request->query('direction') ?? 'asc';
+
+        // Eager loaded products
+        // $products = Product::with('products')->orderBy($sortBy, $direction);
+
+        // Non-eager loaded products
+        $products = Product::query()->orderBy($sortBy, $direction);
```

Comment:
No specification to sort. So, remove that code. There should be eager loading code in your show action though

Reply:
nothing to eager load on the index page

Reply:
for the `show` action, you do need to eager load the reviews for the product